### ISSUE TITLE
[Day 3] Dense 신경망으로 식물 병리 분류 완료 - {이상민}

### ISSUE THREAD

**Issue by tkdals730** [tkdals730/machine-learning#3]
## 실습 완료 내용

- [x] Plant Pathology 2020 데이터 로드 & 탐색

- [x] 이미지 전처리 (224×224 축소, 정규화)

- [x] Dense 신경망 구축 (4-class 분류)

- [x] 클래스 불균형 보정 (class_weight)

- [x] 모델 훈련 (조기 종료 적용)

- [x] 성능 평가 및 혼동행렬 분석

## 데이터셋 정보

- 원본 크기: 1365×1024 이미지

- 전처리: 224×224 축소 (메모리 효율)

- 클래스 분포:
[516  91 622 592]
  - Healthy: 516장

  - Multiple Diseases: 91장

  - Rust: 622장

  - Scab: 592장
  
클래스 분포 (%):
  Healthy: 28.3%
  Multiple Diseases: 5.0%
  Rust: 34.2%
  Scab: 32.5%

## 모델 구축

- 입력층: 150,528 (224×224×3 이미지 flatten)

- 은닉층 1: Dense(512) + BatchNorm + Dropout(0.4) + ReLU

- 은닉층 2: Dense(256) + BatchNorm + Dropout(0.3) + ReLU

- 은닉층 3: Dense(128) + BatchNorm + Dropout(0.2) + ReLU

- 은닉층 4: Dense(64) + ReLU

- 출력층: Dense(4) + Softmax (다중분류)

## 성능 결과

- 훈련 정확도: ____% 

- 검증 정확도: ____% (목표: 70% 이상)

- Log Loss: ____ (낮을수록 좋음)

- 훈련 에포크: 22개 (조기 종료 적용)
- 

<img width="752" height="384" alt="Image" src="https://github.com/user-attachments/assets/e42abf6b-8707-456e-bfdf-ea12ec1892ef" />

<img width="747" height="258" alt="Image" src="https://github.com/user-attachments/assets/650547b6-2b02-448a-b77d-4105eb9d24dd" />

## 혼동행렬 분석 (4×4)

| 실제 \ 예측 | Healthy | Multiple | Rust | Scab |

|-----------|---------|----------|------|------|

| Healthy   | ____ | ____ | ____ | ____ |

| Multiple  | ____ | ____ | ____ | ____ |

| Rust      | ____ | ____ | ____ | ____ |

| Scab      | ____ | ____ | ____ | ____ |

## 클래스별 정확도

클래스별 정확도:
  Healthy: 0.5146 (51.46%)
  Multiple Diseases: 0.2222 (22.22%)
  Rust: 0.4320 (43.20%)
  Scab: 0.4118 (41.18%)

## 관찰 기록

1. 이미지 축소 (1365×1024 → 224×224)가 성능에 미친 영향?

2. 클래스 가중치 적용 전후 비교:

3. 가장 많이 혼동되는 클래스 쌍:

## 개선 시도 (선택)

- 시도 1: (예: 이미지 크기 조정) → 결과: ____%

- 시도 2: (예: Batch Normalization 제거) → 결과: ____%

- 가장 효과 있었던 변경: 

## 다음 단계 (내일)

CNN으로 같은 Plant Pathology 문제를 풀어서 Dense Network와 성능 비교 예정

## 커밋 및 링크

- Kaggle 노트북: (URL)

- GitHub: (링크)
